packages/core/src/components/axes/grid.ts

```diff
@@ -0,0 +1,227 @@
+			.tickSizeOuter(0);
+
+		// Determine number of ticks
+		const numberOfTicks = Tools.getProperty(this.model.getOptions(), "grid", "y", "numberOfTicks");
+		yGrid.ticks(numberOfTicks);
+
+		const g = svg.select(".y.grid")
+			.attr("transform", `translate(0, ${-this.backdrop.attr("y")})`)
+			.call(yGrid);
+
+		this.cleanGrid(g);
+	}
+
+	/**
+	 * Returns the threshold for the gridline tooltips based on the mouse location.
+	 * Calculated based on the mouse position between the two closest gridlines or edges of chart.
+	 */
+	getGridlineThreshold(mousePos) {
+		// use the space between axis grid ticks to adjust the threshold for the tooltips
+		const svg = this.parent;
+
+		// sort in ascending x translation value order
+		const gridlinesX = svg.selectAll(".x.grid .tick").nodes()
+		.sort((a, b) => {
+			return Number(Tools.getTranslationValues(a).tx) - Number(Tools.getTranslationValues(b).tx);
+		});
+
+		// find the 2 gridlines on either side of the mouse
+		let floor = -1; let ceiling;
+		gridlinesX.forEach((line: HTMLElement, i: any) => {
+			if (mousePos[0] >= +Tools.getTranslationValues(line).tx) {
+				floor ++;
+			}
+		});
+		ceiling = (floor + 1 < gridlinesX.length) ? floor + 1 : gridlinesX.length;
+
+		// get the 'step' between chart gridlines
+		const line1 = gridlinesX[floor];
+		const line2 = gridlinesX[ceiling];
+		let lineSpacing;
+
+		// if the mouse is on edge of charts (mouseX < first gridline || mouseX > last gridline)
+		// we can use the chart edge to determind the threshold for the gridlines
+		if (!line1) {
+			// we are between the first gridline and the chart edge
+			lineSpacing = +Tools.getTranslationValues(line2).tx;
+		} else if (!line2) {
+			// we need to use the chart right bounds in case there isnt a domain axis
+			const gridElement = svg.select(".cc-grid rect.chart-grid-backdrop").node();
+			const width = DOMUtils.getSVGElementSize(gridElement).width;
+
+			lineSpacing = width - +Tools.getTranslationValues(line1).tx;
+		} else {
+			// there are two gridlines to use
+			lineSpacing = +Tools.getTranslationValues(line2).tx - +Tools.getTranslationValues(line1).tx;
+		}
+		const { threshold }  = this.model.getOptions().tooltip.gridline;
+		// return the threshold
+		return lineSpacing * threshold;
+	}
+
+	/**
+	 * Returns the active gridlines based on the gridline threshold and mouse position.
+	 * @param position mouse positon
+	 */
+	getActiveGridline(position) {
+		const threshold = Tools.getProperty(this.model.getOptions, "tooltip", "gridline", "threshold") ?
+			Tools.getProperty(this.model.getOptions, "tooltip", "gridline", "threshold") : this.getGridlineThreshold(position);
+		const svg = this.parent;
+
+		const gridlinesX = svg.selectAll(".x.grid .tick")
+		.filter(function() {
+			const translations = Tools.getTranslationValues(this);
+
+			// threshold for when to display a gridline tooltip
+			const bounds = {
+				min: +translations.tx - threshold,
+				max: +translations.tx + threshold };
```

Comment:
Just want to doulbe check what the `+` is for in this case ... if it's a conversion we may want to use `Number()` instead